### PR TITLE
Accounts V2: Wallet Recover

### PR DESCRIPTION
**Pull request by 0xKiwi**
**What type of PR is this?**
Wallet feature addition

**What does this PR do? Why is it needed?**
This PR adds the ability to recover wallets based on their wallet seed phrase. 

Command usage is:
```
Enter a wallet directory: █home/kiwi/.eth2validators/.prysm-wallet-v2
Enter the wallet recovery seed phrase you would like to recover:
✔ Wallet password: *****************█
[2020-07-20 21:17:33]  INFO accounts-v2: Successfully recovered HD wallet and saved configuration to disk. Make a new validator account with ./prysm.sh validator accounts-2 new wallet-path=/home/kiwi/.eth2validators/.prysm-wallet-v2
```

### CODE REVIEW

File: validator/accounts/v2/wallet_recover.go [hunk 1]
```diff
@@ -0,0 +1,113 @@
+package v2
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/manifoldco/promptui"
+
```
Comment (rauljordan) [inline validator/accounts/v2/wallet_recover.go:9]:
```suggestion
```

File: validator/accounts/v2/wallet_recover.go [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+	if strings.Trim(mnemonic, " ") == "" {
+		return errors.New("phrase cannot be empty")
+	}
+	words := strings.Split(mnemonic, " ")
+	for i, word := range words {
+		if strings.Trim(word, " ") == "" {
+			words = append(words[:i], words[i+1:]...)
+		}
+	}
+	if len(words) != 24 {
```
Comment (rauljordan) [inline validator/accounts/v2/wallet_recover.go:109]: Make this 24 a const

File: validator/keymanager/v2/derived/derived.go [hunk 3]
```diff
@@ -10,6 +10,8 @@ import (
 	"strconv"
 	"sync"
 
+	"github.com/tyler-smith/go-bip39"
+
```
Comment (rauljordan) [inline validator/keymanager/v2/derived/derived.go:14]:
```suggestion
```

File: validator/accounts/v2/new.go [hunk 4]
```diff
@@ -130,7 +130,21 @@ func inputKeymanagerKind(cliCtx *cli.Context) (v2keymanager.Kind, error) {
...
-func inputNewWalletPassword() (string, error) {
+func inputNewWalletPassword(cliCtx *cli.Context) (string, error) {
+	if cliCtx.IsSet(flags.PasswordFileFlag.Name) {
+		passwordFilePath := cliCtx.String(flags.PasswordFileFlag.Name)
+		data, err := ioutil.ReadFile(passwordFilePath)
+		if err != nil {
+			return "", err
+		}
+		enteredPassword := string(data)
+		fmt.Println(enteredPassword)
```
Comment (rauljordan) [inline validator/accounts/v2/new.go:141]:
```suggestion
```

File: validator/accounts/v2/new.go [hunk 5]
```diff
@@ -252,7 +266,7 @@ func inputPasswordsDirectory(cliCtx *cli.Context) string {
 	}
 	passwordsPath, err := prompt.Run()
 	if err != nil {
-		log.Fatalf("Could not determine passwords directory: %v", formatPromptError(err))
+		log.Errorf("Could not determine passwords directory: %v", formatPromptError(err))
```
Comment (rauljordan) [inline validator/accounts/v2/new.go:268]: Why not fatal

File: validator/accounts/v2/wallet_recover.go [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+	if err != nil {
+		log.Fatal(err)
+	}
+	if walletExists {
+		log.Fatal(
+			"You already have a wallet at the specified path. You can " +
+				"edit your wallet configuration by running ./prysm.sh validator wallet-v2 edit",
+		)
+	}
+	if err = recoverDerivedWallet(cliCtx, walletDir); err != nil {
```
Comment (rauljordan) [inline validator/accounts/v2/wallet_recover.go:39]: We need to make sure this is a derived wallet in the first place

File: validator/accounts/v2/new.go (hunk 4 above)
Comment (0xKiwi) [reply validator/accounts/v2/new.go:141]: lol whoops was getting the not strong enough pass error

File: validator/accounts/v2/wallet_recover.go [hunk 7]
```diff
@@ -0,0 +1,131 @@
...
+		return errors.New("phrase cannot be empty")
+	}
+	words := strings.Split(mnemonic, " ")
+	for i, word := range words {
+		if strings.Trim(word, " ") == "" {
+			words = append(words[:i], words[i+1:]...)
+		}
+	}
+	if len(words) != phraseWordCount {
+		return fmt.Errorf("phrase must be 24 words, entered %d", len(words))
```
Comment (rauljordan) [inline validator/accounts/v2/wallet_recover.go:128]: Also use fmt for the 24 in here with %d

File: validator/keymanager/v2/derived/derived.go [hunk 8]
```diff
@@ -196,6 +197,30 @@ func InitializeWalletSeedFile(ctx context.Context, password string, skipMnemonic
 	}, nil
 }
 
+// SeedFileFromPhrase --
```
Comment (rauljordan) [inline validator/keymanager/v2/derived/derived.go:200]: Wrong comment, let's add an informative one

File: validator/accounts/v2/wallet_recover.go (hunk 6 above)
Comment (0xKiwi) [reply validator/accounts/v2/wallet_recover.go:39]: Theres no existing wallet if you're using recover no? The function already rejects if a wallet exists.

**Pull request merged**